src/evo/providertx.cpp

```diff
@@ -76,6 +76,16 @@ bool CheckProRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValid
     if (!ptx.scriptPayout.IsPayToPublicKeyHash())
         return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
 
+    CTxDestination payoutDest;
+    if (!ExtractDestination(ptx.scriptPayout, payoutDest)) {
+        // should not happen as we checked script types before
+        return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
+    }
+    // don't allow reuse of keys for different purposes
```

Comment:
The comment here is a bit confusing imo. This doesn't allow reuse of keys which control payout address for other purposes, not reuse of keys in general.